tests/unit/onyxMultiMergeWebStorageTest.js

```diff
@@ -0,0 +1,70 @@
+import OnyxCache from '../../lib/OnyxCache';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+import localforageMock from '../../__mocks__/localforage';
+
+const ONYX_KEYS = {
+    COLLECTION: {
+        TEST_KEY: 'test_',
+    },
+};
+
+const initialTestObject = {a: 'a'};
+const initialData = {
+    test_1: initialTestObject,
+    test_2: initialTestObject,
+    test_3: initialTestObject,
+};
+
+localforageMock.setInitialMockData(initialData);
+
+describe('Onyx.mergeCollection()', () => {
+    let Onyx;
+    let Storage;
+    beforeAll(() => {
+        jest.mock('../../lib/storage');
+        Onyx = require('../../index').default;
+        Storage = require('../../lib/storage').default;
+        jest.useRealTimers();
+    });
+
+    it('mergeCollection', () => {
+        expect(Storage.name).toBe('WebStorage');
```

Comment:
Does the success of the test depend on the underlying storage provider? - it shouldn't 
`mergeCollection` should be considered working correctly if it correctly delegates work to a Storage provider

Testing that `mergeCollection` is interacting correctly with cache is valuable

Reply:
Since the implementation we are having issues with is the storage provider then I want to make sure things work correctly - it also took me some time to get this working. Testing against AsyncStorage would not touch any of the code we're proposing to add. Sorry, I think you are trying to say that we should maybe:

1. Have one set of tests
2. Run them against both storage providers

But not totally sure. Please clarify if there is a specific suggestion, thanks!

Reply:
Actually I don't think it would be worth it to run against all storage providers

We've introduced 2 changes: 
1. Queue LocalForage writes - this can be tested outside of `mergeCollection`
2. We changed the way `meregeCollection` updates cache - this shouldn't depend on specific storage provider

Specifically for the bug with partial cache I think the issue should be happening with any storage provider, because we evicted cache keys